src/fetchtastic/download/client_app.py

```
@@ -65,6 +81,62 @@
+
+    flavor: str
+    abi: str | None  # str=exact, _ABI_WILDCARD=any, None=default(universal)
+
+
+# Legacy generic official APK names that don't carry a flavor token.
+_LEGACY_GENERIC_APK_NAMES = frozenset(
+    {"meshtastic.apk", "app-release.apk", "app.apk", "meshtastic-app.apk"}
+)
+
+
+def _parse_apk_identity(name: str) -> ApkIdentity | None:
```

Ruff (0.15.20)

[warning] Too many return statements (7 > 6)

(PLR0911)